common/src/main/java/io/novafoundation/nova/common/presentation/AssetIconProvider.kt

```diff
@@ -0,0 +1,44 @@
+package io.novafoundation.nova.common.presentation
+
+import io.novafoundation.nova.common.R
+import io.novafoundation.nova.common.data.model.AssetIconMode
+import io.novafoundation.nova.common.data.repository.AssetsIconModeRepository
+import io.novafoundation.nova.common.utils.images.Icon
+import io.novafoundation.nova.common.utils.images.asIcon
+
+interface AssetIconProvider {
+
+    val fallbackIcon: Icon
+
+    fun getAssetIcon(iconName: String?, fallback: Icon = fallbackIcon): Icon
+
+    fun getWhiteAssetIcon(iconName: String?, fallback: Icon = fallbackIcon): Icon
```

Comment:
I gave a few suggestions for the interface here:
1. The idea of having fallback icon being exposed by AssetIconProvider looks good. However I think we can make the the interface clearer if `getAssetIcon` wont allow for nullable `iconName`. Instead, we can create an extension function that allows nullability and has the same signature that is currently present in the interface directly. This way we will simplify the interface itself
2. Instead of creating a separate instance of `getWhiteAssetIcon` how about we make a function `getAssetIcon(iconName, iconType)` and implement`getWhiteAssetIcon` in a form of extension? This way you wont need to icon formatting logic twice since `getAssetIcon(iconName)` can easily delegate its work to `getAssetIcon(iconName, iconType)` by passing `iconType=selectedIconType`